stdlib/Distributed/src/managers.jl

```diff
@@ -725,3 +725,26 @@ function kill(manager::SSHManager, pid::Int, config::WorkerConfig)
     cancel_ssh_tunnel(config)
     nothing
 end
+
+function kill(manager::LocalManager, pid::Int, config::WorkerConfig; exit_timeout = 15, term_timeout = 15)
+    # First, try sending `exit()` to the remote over the usual control channels
+    remote_do(exit, pid)
+
+    timer_task = @async begin
+        sleep(exit_timeout)
+
+        # Check to see if our child exited, and if not, send an actual kill signal
+        if !process_exited(config.process)
+            @warn("Failed to gracefully kill worker $(pid), sending SIGTERM")
+            kill(config.process, Base.SIGTERM)
```

Comment:
Is it worth also before this triggering the new 1 second profiler? SIGINFO on BSDs, SIGUSR1 on others

Reply:
It probably needs at least 2-3 seconds to run and print.

Reply:
Can you open a new PR about that?  We'll need to ensure that it doesn't, for instance, output to `stdout` and corrupt otherwise useful output from a worker.